### PR TITLE
Changes to export code to google3.

### PR DESCRIPTION
**Pull request by andychu**
- Bug fix: if the map cache .rda file can't be written, log an error rather
than stopping analysis
- bin/test.sh: for decode-dist and decode-assoc smoke tests, write the testdata
in a layout that is more easily exported
- add DEP_PYTHON and DEP_FAST_EM environment variables
- scripts/g3export: script to export analysis code and generated testdata

### CODE REVIEW

File: scripts/g3export.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#   // Export data to rappor_analysis/testdata
+#   scripts/g3export.sh testdata ~/foo/google3
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+readonly THIS_DIR=$(dirname $0)
+readonly RAPPOR_SRC=$(cd $THIS_DIR/.. && pwd)
+readonly EM_CPP_EXECUTABLE=$RAPPOR_SRC/analysis/cpp/_tmp/fast_em
```
Comment (ilyamironov) [inline scripts/g3export.sh]: Is this constant used anywhere?

File: scripts/g3export.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+testdata() {
+  local dest_g3=$1
+  ensure-google3 $dest_g3
+
+  local dest=$dest_g3/third_party/rappor_analysis/generated_testdata
+
+  local this_script=$RAPPOR_SRC/scripts/g3export.sh
+
+  # Run from the _tmp dir to get the right relative paths.
+  pushd $RAPPOR_SRC/_tmp
```
Comment (ilyamironov) [inline scripts/g3export.sh]: _tmp was parameterized elsewhere. Do you want to define it as constant?

**Comment by andychu** [comment]
Ilya, Tom wanted me to  move g3export.sh to the internal repo, not external.  So I will have a separate change with those changes.

**Comment by tkaitchuck** [comment]
LGTM

**Pull request merged**